### PR TITLE
#7716: Created form for API key, validation for this form

### PR DESCRIPTION
**Pull request by aster06**

### CODE REVIEW

**Review by ivandudardb — APPROVED** [review]
LGTM

File: composer.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
         "drupal/linked_field": "^1.5",
         "drupal/menu_block": "^1.10",
         "drupal/menu_item_extras": "^3.0",
+        "drupal/menu_per_role": "^1.6",
```
Comment (lavrenchuk-dev) [inline composer.json:31]: Can you explain why this module is needed?

File: config/default/core.menu.static_menu_link_overrides.yml [hunk 2]
```diff
@@ -8,15 +8,15 @@ definitions:
     expanded: false
     enabled: true
   user__page:
-    weight: -46
+    enabled: false
```
Comment (lavrenchuk-dev) [inline config/default/core.menu.static_menu_link_overrides.yml:11]: Not sure why you are disabling user__page link

File: web/modules/custom/weather/config/install/weather.settings.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+api_admin_key: "Api key."
```
Comment (lavrenchuk-dev) [inline web/modules/custom/weather/config/install/weather.settings.yml]: No really need to provide default configuration if it doesn't do anything.

**Review by lavrenchuk-dev — CHANGES_REQUESTED** [review]

**Review by lavrenchuk-dev — APPROVED** [review]
Remove menu_per_role from the composer as well in your next PR.

**Pull request merged**